core/src/main/java/com/datastax/oss/driver/api/core/connection/ReconnectionPolicy.java

```diff
@@ -15,23 +15,41 @@
  */
 package com.datastax.oss.driver.api.core.connection;
 
+import com.datastax.oss.driver.api.core.metadata.Node;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.time.Duration;
 
 /**
- * Decides how often the driver tries to re-establish lost connections to a node.
+ * Decides how often the driver tries to re-establish lost connections.
  *
- * <p>Each time a connection to a node is lost, a {@link #newSchedule() new schedule} instance gets
- * created. Then {@link ReconnectionSchedule#nextDelay()} will be called each time the driver needs
- * to schedule the next connection attempt. When the node is back to its required number of
- * connections, the schedule will be reset (that is, the next failure will create a fresh schedule
- * instance).
+ * <p>Each time a reconnection starts, a new {@link ReconnectionSchedule} instance gets created.
+ * Then {@link ReconnectionSchedule#nextDelay()} will be called each time the driver needs to
+ * program the next connection attempt. When the reconnection succeeds, the schedule is discarded,
+ * and the next reconnection will start from a fresh instance.
```

Comment:
I found myself slightly preferring the old rephrasing, would you mind adding a bit more context: "When the reconnection succeeds, the schedule is discarded; if the connection is lost again, the next reconnection attempt will query the policy again to obtain a new schedule".